Orange/widgets/data/owfeatureconstructor.py

```diff
@@ -663,6 +652,68 @@ def freevars(exp, env):
         raise ValueError(exp)
 
 
+def validate_exp(exp):
+    """
+    Validate an `ast.AST` expression.
+
+    Only expressions with no list,set,dict,generator comprehensions
```

Comment:
> I've been meaning to introduce a ListVariable type. Seems like I wouldn't be able to populate it with feature constructor?

No, unless you enable/change the accepted syntax for your variable type.

> I see other uses cases where the expression evaluated to a list of values would be useful (like setting the values of all instances at once (though for that I'd also need to be given all instances at once))

That should be done by a vectorized expression, which is an argument for limiting the syntax further.